### PR TITLE
Clean Code: replace equal("") with BeEmpty()

### PR DESCRIPTION
**Pull request by nunnatsa**
**What this PR does / why we need it**:
Use the dedicated ginkgo matcher `BeEmpty()` for empty strings, instead of using `equal("")`, in order to get better error description and more readable test code.

**Release note**:

```release-note
None
```

### CODE REVIEW

**Review by maiqueb — APPROVED** [review]
Thank you.

**Comment by maiqueb** [comment]
only thing I wonder is if we really need 26 commits to change this. All in one, its one change.

**Comment by acardace** [comment]
> only thing I wonder is if we really need 26 commits to change this. All in one, its one change.

I agree, it seems a bit overkill, you can probably just squash everything into 1 commit and add the command you used to do the replacing (if any) in the commit msg.

**Comment by nunnatsa** [comment]
> > only thing I wonder is if we really need 26 commits to change this. All in one, its one change.
> 
> I agree, it seems a bit overkill, you can probably just squash everything into 1 commit and add the command you used to do the replacing (if any) in the commit msg.

@acardace @maiqueb - fixed.

**Review by enp0s3 — APPROVED** [review]
/approve

File: pkg/virt-operator/resource/apply/core_test.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ var _ = Describe("Apply", func() {
 
 	Context("Services", func() {
 
-		It("should patch if ClusterIp == \"\" during update", func() {
+		It(`should not patch if ClusterIp == "" during update`, func() {
```
Comment (EdDev) [inline pkg/virt-operator/resource/apply/core_test.go:57]: This change (and there are other I think), do not fit the same commit (or even PR) as the subject.
Can you extract it to a different PR?

**Review by EdDev — CHANGES_REQUESTED** [review]
> Use the dedicated ginkgo matcher BeEmpty() for empty strings, instead of using equal(""), in order to get better error description and more readable test code.

If the equality is on strings, I do not understand what better description will the `BeEmpty` give here. When you mention "more readable test code", what do you mean exactly? Why is it better reading that an explicit empty string?
I think you need to elaborate a bit to convince this is indeed better.

I can give here a counterpart reasoning why this may not be better: Using `Equal` is strict about types (at least per the documentation) while `BeEmpty` is not.

I usually use `BeEmpty` for checking arrays, slices and maps, as these may come as "empty" or `nil`, both valid in terms of usage in code (and interchangeable most of the time).
Still, we do loose the type checking, but in those cases I think it is worth it.
I am not that sure if it is valid for strings as well.

I would also recommend splitting this into two commits: One for the production code and one for the e2e tests.
